pkg/api/customization/clustertemplate/cluster_template.go

```diff
@@ -33,6 +39,17 @@ func (w Wrapper) Formatter(apiContext *types.APIContext, resource *types.RawReso
 	resource.Links["revisions"] = apiContext.URLBuilder.Link("revisions", resource)
 }
 
+func (w Wrapper) RevisionFormatter(apiContext *types.APIContext, resource *types.RawResource) {
+
+	if err := apiContext.AccessControl.CanDo(v3.ClusterTemplateRevisionGroupVersionKind.Group, v3.ClusterTemplateRevisionResource.Name, "update", apiContext, resource.Values, apiContext.Schema); err == nil {
+		if convert.ToBool(resource.Values["enabled"]) {
+			resource.AddAction(apiContext, disableRevisionAction)
```

Comment:
do we need to check if it's actually in use before allowing disable action?

Reply:
No we are allowing to disable even if the revision is in use by some clusters. After disabling the revision is not available for new cluster creation.